docs/src/appendix/packaging.md

```diff
@@ -0,0 +1,501 @@
+representation. This means that every 8 bytes will be interpreted as the canonical little-endian byte representation
+of a single field element.
+
+In both cases, the data written into linear memory will be padded with zeros to the nearest word boundary. For example,
+if you have 13 elements worth of data (3 words + 1 element), then the actual segment size will be 16 elements, with
+the last 3 all zeroes.
+
+Segment overlap is not permitted, except when the overlap only occurs due to padding. Segments are written in increasing
+address order, so padding bytes will never overwrite non-padding bytes.
+
+> [!IMPORTANT]
+> The `[rodata]` section is only used for creating a package from a directory.
+> It will never be present in a manifest extracted from a package, because the
+> rodata will have been emitted in canonical on-disk form in the `rodata` directory
+
+#### Extras
+
+Certain items that a package can contain are considered optional extras. These items can be
+controlled from the `[extras]` section:
+
+```toml
+[extras]
+docs = true
+debug = true
+```
+
+The entire `[extras]` section is optional, in which all optional features are assumed to be disabled.
+If the section is present, any items that are not specified, are assumed to be disabled.
+
+The structure and full configuration of extras is not yet fully defined, but the following are some
+initial thoughts:
+
+* The `docs` field indicates whether or not to include documentation in the package. If `true`, then
+the docs are expected to be in the form of a static HTML site, placed under the `etc/docs` directory,
+with an `index.html` file as the main page of the docs. The docs will be compressed as a zip file,
+and stored within the package. If `false`, then whether or not `etc/docs` is present doesn't matter,
+it will not be included in the package binary.
+
+* The `debug` field indicates whether or not to include source-level debugging information in the package.
+If `true`, then the debug info/source maps/etc., will be expected under `etc/debug`. Currently, there is
+no format for this data, so this setting has no effect.
+
+#### Dependencies
+
+The final section of the manifest, is `[dependencies]`. This section specifies what dependencies this
+package has, and optionally, how to fetch them.
+
+The most basic way to express a dependency is as follows:
+
+```toml
+[dependencies]
+foo = "0.1.0"
+```
+
+This expresses a dependency on the `foo` package at version `0.1.0`. When expressed this way, any suitable
+match for this version string can satisfy this dependency. As such, the package must not contain direct
+references to specific MAST roots in `foo`, as the roots are not guaranteed to be stable across builds.
+For certain things, this flexibility may be convenient. However, if you want to be more precise, you
+can specify the content digest to match:
+
+```toml
+[dependencies]
+foo = { version = "0.1.0", digest = "..." }
```

Comment:
I don't think I was expecting both to be present at the same time, but I don't see any reason why we couldn't support looking up a package by digest alone.

I was initially expecting that a `version` would be used without a `digest`, and is equivalent of stipulating "as long as you have a package that matches this version requirement, I don't care what the digest is". However, after doing the assembler refactoring, I don't think it will ever be the case that we will be referencing code by anything other than MAST digest, so `version` is effectively meaningless (it is perhaps useful as a way of answering the question "what semantic version does this digest correspond to?", but that's about it, and could be provided via other means).

I'll mull this over a bit just to try and confirm that I'm not forgetting some detail that I had in mind when I drafted this, but I think we may be able to remove `version` from the package manifest.